### PR TITLE
 #168131430 Notification for approved request

### PR DESCRIPTION
**Pull request by Dkabiswa**
#### What does this PR do?
Integrates sending a notification for both approved and rejected requests feature
#### Description of Task to be completed?
- Notify user when their request is approved or rejected
#### How should this be manually tested?
- To test the above feature locally you can get info on the setup from this [repo](https://github.com/drayzii/notifications) being used for notifications
- Git clone this repo and checkout to this branch
- Run `npm install`, `npm run db-migrate:dev` and `npm run db-seed:test`
- Run `npm run start:dev` to start server
- Open postman and login as a manager
- POST `http://localhost:3000/api/v1/auth/signin`
```
{
"userEmail": marveldev53@gmail.com
"userPassword": Root1123#
}
```
- Create a request
- POST `http://localhost:3000/api/v1/auth/signin`
```
{
"from":"Kigali, Rwanda",
"to": 1,
"travelDate": "2019-10-18",
"reason": "iam travelling cause the company allows us to, i mean the company finances everything so why not",
"accommodation": "hotel"
}
```
- Approve or Reject a request
- `http://localhost:3000/api/v1/requests/approve/:requestId`
- `http://localhost:3000/api/v1/requests/reject/:requestId`
- Look out for a push notification
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#168131430](https://www.pivotaltracker.com/story/show/168131430)
#### Screenshots (if appropriate)
Screen Shot 2019-10-03 at 15 33 26
Screen Shot 2019-10-03 at 15 32 42

#### Questions:
N/A

### CODE REVIEW

File: src/utils/notifications/requestNotification.js [hunk 1]
```diff
@@ -59,6 +72,23 @@ class Notifications {
...
+    await emitter.on('request update', async (request) => {
+      let notificationMessage;
+      let ownerId;
+      const { userId } = await this.makeNotification(request);
+      if (request.status !== 'Pending') {
+        notificationMessage = `Your Request has been ${request.status.toLowerCase()}`;
+        ownerId = userId;
+      }
+
+      this.notify(notificationMessage, ownerId, 'updated', request.id);
```
Comment (drayzii) [inline src/utils/notifications/requestNotification.js]: Shouldn't this line be inside the if statement above

File: src/utils/notifications/requestNotification.js [hunk 2]
```diff
@@ -59,6 +72,23 @@ class Notifications {
...
+    });
+  }
```
Comment (drayzii) [inline src/utils/notifications/requestNotification.js]: I think we should move the emailing part of the feature to this function and check if the user hasn't opted out of emails. Or add the check to the controller

File: src/utils/notifications/requestNotification.js (hunk 1 above)
Comment (Dkabiswa) [reply src/utils/notifications/requestNotification.js]: I left it out cause there is another feature which is to use the same method and it does not use the check

File: src/utils/notifications/requestNotification.js (hunk 2 above)
Comment (Dkabiswa) [reply src/utils/notifications/requestNotification.js]: that has to be in the controller cause it was not part of this story

**Review by drayzii — APPROVED** [review]

**Review by Bahatiroben — APPROVED** [review]
looks good

**Review by Paulvitalis200 — APPROVED** [review]

**Pull request merged**